### PR TITLE
HParam: Implement new Data Table projection architecture in Runs Table

### PR DESCRIPTION
**Pull request by JamesHollyer**
## Motivation for features / changes
We preparation for adding HParam functionality to both the Scalar Table and the Runs Table we have refactored the Data Table to be more flexible. This refactoring was done in #6422 and #6427 and was only implemented in the Scalar Table because the Runs Table only uses the Data Table behind a flag. This PR implements the new Data Table Structure in the Runs Table.

## Technical description of changes
Basically just followed the same structure as the Scalar Table.

## Screenshots of UI changes (or N/A)
Runs Table specific styling still needs to be done. It currently looks like this.
Screenshot 2023-06-15 at 7 21 46 PM

We still need to add a column for check boxes and add styling.

### CODE REVIEW

File: tensorboard/webapp/runs/views/runs_table/runs_table_container.ts [hunk 1]
```diff
@@ -180,7 +180,7 @@ function matchFilter(
   }
   if (filter.type === DomainType.DISCRETE) {
     // (upcast to work around bad TypeScript libdefs)
-    const values: Readonly<Array<typeof filter.filterValues[number]>> =
+    const values: Readonly<Array<(typeof filter.filterValues)[number]>> =
```
Comment (JamesHollyer) [inline tensorboard/webapp/runs/views/runs_table/runs_table_container.ts:183]: I got a link error in CI. I ran yarn fix-lint. This was the change. ¯\(ツ)/¯

File: tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    fixture.detectChanges();
+    return fixture;
+  }
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [DataTableModule, MatIconTestingModule],
+      declarations: [TestableComponent, RunsDataTable],
+      //   schemas: [NO_ERRORS_SCHEMA],
```
Comment (rileyajones) [inline tensorboard/webapp/runs/views/runs_table/runs_data_table_test.ts:99]:
```suggestion
```

**Review by rileyajones — APPROVED** [review]

File: tensorboard/webapp/runs/views/runs_table/runs_data_table.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+==============================================================================*/
+$_circle-size: 12px;
+
+.row-circle {
+  height: $_circle-size;
+  width: $_circle-size;
+}
+.row-circle > span {
+  border-radius: 50%;
+  border: 1px solid rgba(255, 255, 255, 0.4);
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/runs/views/runs_table/runs_data_table.scss:23]: This is how it is done in many places. I am leaving it for now. Maybe I will come do a broad change for all circle icons.
